platform-server-kernel/src/main/java/uno/cod/platform/server/core/domain/Task.java

```diff
@@ -35,5 +45,47 @@ public void setName(String name) {
     protected void setChallenges(List<Challenge> challenges) {
         this.challenges = challenges;
     }
+
+    public Endpoint getEndpoint() {
+        return endpoint;
+    }
+
+    public void setEndpoint(Endpoint endpoint) {
+        this.endpoint = endpoint;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public String getInstructions() {
+        return instructions;
+    }
+
+    public void setInstructions(String instructions) {
+        this.instructions = instructions;
+    }
+
+    public Organization getOrganization() {
+        return organization;
+    }
+
+    public void setOrganization(Organization organization) {
+        this.organization = organization;
+    }
+
+    protected void addChallenge(Challenge challenge) {
+        if (challenge == null) {
+            throw new IllegalArgumentException("challenge not valid");
+        }
+        if (challenges == null) {
+            challenges = new ArrayList<>();
+        }
+        challenges.add(challenge);
```

Comment:
as you see, the method is protected, and the only place where we call that is from Challenge.addTask so we would have an infinite loop. This is the logic way. We create a challenge -> we add tasks -> we set the challenge to the tasks